src/ripple/app/ledger/impl/LocalTxs.cpp

```diff
@@ -150,8 +150,11 @@ class LocalTxsImp
                 return true;
             if (view.txExists(txn.getID()))
                 return true;
-            auto const sle = cachedRead(view,
-                keylet::account(txn.getAccount()).key, ltACCOUNT_ROOT);
+
+            auto const& key = keylet::account(txn.getAccount()).key;
+            std::shared_ptr<SLE const> sle = ltACCOUNT_ROOT ?
+                view.read(Keylet{ltACCOUNT_ROOT, key}) :
+                view.read(keylet::unchecked(key));
```

Comment:
In this instance we _know_ the type of the keylet we want (`ltACCOUNT_ROOT`) so we don't need the ternary at all. Lines 154-157 can just be reduced to:

    auto const sle = view.read(keylet::account(txn.getAccount()));

Reply:
Ah, I now see what you meant by that comment in the issue. It makes sense now, and like you pointed out, applies in all the cases where we inlined this function. 

I will fix them up tonight and also reword the commit so that it auto closes the issue when this is merged. Thanks, @nbougalis!

Reply:
Thanks! Your refactor was basically exactly what I described, so it's on me. I should have been more clear.